File: SchoolAngular/src/app/service/group.schedule.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import {GroupSchedule} from "../model/group.schedule.model";
import {GroupScheduleView} from "../model/group.schedule.view.model";
import {GroupScheduleStudentView} from "../model/group.schedule.student.view.model";

@Injectable()
export class GroupScheduleService {

  private basUrl = "http://localhost:8080/groupschedule"

  constructor(private httpClient: HttpClient) {

  }

  getGroupScheduleList(): Observable<GroupSchedule[]> {
    return this.httpClient.get<GroupSchedule[]>(`${this.basUrl}`);
  }

  getGroupScheduleListForGroup(groupId: number): Observable<GroupSchedule[]> {
    return this.httpClient.get<GroupSchedule[]>(`${this.basUrl}/find-for-group/${groupId}`);
  }

  createGroupSchedule(groupSchedule: GroupSchedule): Observable<GroupSchedule> {
    return this.httpClient.post<GroupSchedule>(`${this.basUrl}`, groupSchedule);
  }

  getGroupScheduleById(id: number): Observable<GroupSchedule>{
    return this.httpClient.get<GroupSchedule>(`${this.basUrl}/${id}`);
  }

  updateGroupSchedule(id:number, groupSchedule:GroupSchedule): Observable<Object>{
    return this.httpClient.put(`${this.basUrl}/${id}`, groupSchedule);
  }

  deleteGroupSchedule(id:number): Observable<Object>{
    return this.httpClient.delete(`${this.basUrl}/${id}`);
  }

  findGroupScheduleViewListByTeacherId(teacherId: number): Observable<GroupScheduleView[]> {
    return this.httpClient.get<GroupScheduleView[]>(`${this.basUrl}/find-for-teacher/${teacherId}`);
  }

  findGroupScheduleViewListByStudentId(studentId: number): Observable<GroupScheduleStudentView[]> {
    return this.httpClient.get<GroupScheduleStudentView[]>(`${this.basUrl}/find-for-student/${studentId}`);
  }

  findGroupScheduleViewById(id: number): Observable<GroupScheduleView> {
    return this.httpClient.get<GroupScheduleView>(`${this.basUrl}/find-view-by-id/${id}`);
  }


}

File: springCurseWork/src/main/java/com/example/springcursework/servise/GroupScheduleService.java
package com.example.springcursework.servise;

import com.example.springcursework.model.GroupSchedule;
import com.example.springcursework.model.GroupScheduleStudentView;
import com.example.springcursework.model.GroupScheduleView;

import java.util.List;

public interface GroupScheduleService {
    public GroupSchedule insert(GroupSchedule groupScheduleVO);
    public List<GroupSchedule> findAll();
    public List<GroupSchedule> findGroupScheduleListForGroup(int groupId);
    public void delete(int id);
    public GroupSchedule findById(int id);
    public GroupSchedule updateGroupSchedule(int id, GroupSchedule groupScheduleVO);
    public List<GroupScheduleView> findByTeacherId(int teacherId);
    public GroupScheduleView findGroupScheduleViewById(int id);
    public List<GroupScheduleStudentView> findByStudentId(int studentId);
}
